### PR TITLE
Allow schema (B, A -> B, B) for recommendation request

### PR DESCRIPTION
**Pull request by kubasimon**
(interaction user is same as target user on merge)

Addressing #63

### CODE REVIEW

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (OndraM) [inline tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: Maybe there is opportunity for data provider ($interaction, $userMerge, $recommendation) and just extend the previous testcase, what do you think?

File: src/RequestBuilder/RecommendationRequestBuilder.php
Comment (OndraM) [inline src/RequestBuilder/RecommendationRequestBuilder.php]: Maybe move this after the following condition, so that cases for "not null user-merge" are next to each other?

**Review by OndraM — CHANGES_REQUESTED** [review]

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (kubasimon) [reply tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: ok, changed to use same test with dataprovider

File: src/RequestBuilder/RecommendationRequestBuilder.php
Comment (kubasimon) [reply src/RequestBuilder/RecommendationRequestBuilder.php]: moved

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (OndraM) [inline tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: The comment could instead be used as array key, its then printed in PHPUnit output, like:

```php
'(A, A -> B, B)' => ['test-user-a', 'test-user-a', 'test-user-b', 'test-user-b'],
```

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (OndraM) [inline tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: - could you please add return typehint?
- also please add phpdoc to help phpstorm & phpstan recognize internal type of array contents
- naming convence for provider method name is "provideFooBar", eg. to not conincidentally name them like "testUsersProvider", which would cause PHpUnit to recognize them as tests and do unexpected things

So this should be:
```php
     /**
     * @return array[]
     */
    public function providecorrectSequenceOfUsers(): array
```

**Review by OndraM — CHANGES_REQUESTED** [review]
Thanks, just to really minor things. I also asked @foglcz to have a quick look as well, but I hope we can merge this then soon - meaning today :).

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (kubasimon) [reply tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: done

File: tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php
Comment (kubasimon) [reply tests/unit/RequestBuilder/RecommendationRequestBuilderTest.php]: done

**Review by OndraM — APPROVED** [review]

**Comment by OndraM** [comment]
Thanks @tenerd ! Let's wait for @foglcz and I will merge it then if he finds no issues :)

**Review by foglcz — APPROVED** [review]

**Pull request merged**
